### PR TITLE
KT-23559: Fix hint text for ReplaceWithOperatorAssignmentInspection

### PR DESCRIPTION
**Pull request by Mon-chi**

### CODE REVIEW

**Comment by Mon-chi** [comment]
https://youtrack.jetbrains.com/issue/KT-23559

File: idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceWithOperatorAssignmentInspection.kt [hunk 1]
```diff
@@ -55,7 +55,7 @@ class ReplaceWithOperatorAssignmentInspection : AbstractApplicabilityBasedInspec
     override val defaultFixText = "Replace with operator-assignment"
 
     override fun fixText(element: KtBinaryExpression) =
-            "Replace with '${element.operationReference.operationSignTokenType?.value}='"
+        "Replace with '${(element.right as KtBinaryExpression).operationReference.operationSignTokenType?.value}='"
```
Comment (mglukhikh) [inline idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceWithOperatorAssignmentInspection.kt]: Just for safety, I'd use `as?` instead of `as` here

Comment (mglukhikh) [reply idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceWithOperatorAssignmentInspection.kt]: Also, some test should be provided. Please look at `idea/testData/inspectionsLocal/replaceWithOperatorAssignment` for examples. You should use "Generate All Tests" run configuration to regenerate automatic tests (`LocalInspectionTestGenerated` in our case).

Comment (mglukhikh) [reply idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceWithOperatorAssignmentInspection.kt]: Also I recommend to run other local inspection tests to be sure that nothing is broken

**Comment by Mon-chi** [comment]
I'm sorry for late reply. I was too busy to work on this.

What should be happen when element.right !as ktBinaryExpression?
And I can't use Generate All Tests due to the bug with Windows. What should I do?

**Comment by mglukhikh** [comment]
> What should be happen when element.right !as ktBinaryExpression?

Nothing special, just insert safe calls in this expression (fixText) everywhere.

**Comment by mglukhikh** [comment]
> And I can't use Generate All Tests due to the bug with Windows. What should I do?

Yes, we had an error at this point with a message like "too long command line". However, it's already fixed. You can rebase your branch on master and everything should work properly (I use Windows myself).

**Comment by Mon-chi** [comment]
Thank you for your support. I've updated.

File: idea/testData/inspectionsLocal/replaceWithOperatorAssignment/validAddition.kt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+fun foo() {
```
Comment (mglukhikh) [inline idea/testData/inspectionsLocal/replaceWithOperatorAssignment/validAddition.kt]: This test does not test your change yet. You should add comment above (to both this file and after file):

```
// FIX: expected fix text
```

to check that fix text is correct.

**Comment by Mon-chi** [comment]
Updated. Is this alright?

**Comment by mglukhikh** [comment]
Yes, many thanks. PR accepted, I'll merge it this week.

**Comment by mglukhikh** [comment]
Merged, thank you

**Pull request closed without merging**